Aa search

adrianAlday opened the pull request:
#11 

Adding search capability:

Search goes through name, themes, blurb, and both number formats
Search updates debounced to 300ms
When not searching, list sorted by short name, except majors go first 
When searching, list sorted by search score, best always first

fmapilis commented on components/displayedCardList.js:
```diff
@@ -0,0 +1,51 @@
+
+export default connect(
+  mapStateToProps,
+)( cardList );
```
This isn't actually a component, so we shouldn't put this in the `/components` folder.

We also generally don't want to move the `mapStateToProps` logic outside of the component that's actually using it (which in this case is `CardList`). We want to keep that all in the same place so that it's easier to tell what `props` are coming into `CardList` and why.

The logic in this file would belong better in a selector using the `reselect` library. We can go over that on Monday if you want some help on it.

fmapilis commented on pages/cards.js:
```diff
@@ -1,27 +1,32 @@
   }
 
-  handleSearchFocus = () => {
+  debouncedSendSearchQuery = _.debounce(( eventVal ) => {
```
We'll probably want to use `_.throttle` instead of `_.debounce` here.

Debouncing doesn't execute the debounced function unless x milliseconds elapsed without it having been called again. Let's say we're debouncing by 300ms, and a person is typing out a search query that takes 1500ms, but they're typing fast enough that 300ms doesn't go by without them entering in a new letter. The debounced function won't get called until  300ms elapses after the last thing they enter, which in this case would be at 1800ms.

Throttling will execute _at most_ once every x milliseconds but guarantees _at least_ one execution in that time frame as long as it was called at least once. Using the same example above, this would execute every 300ms, which is a better "live updating" search experience.

adrianAlday commented:
In c6febbe, added new icons for readings and suits, changed corner icon size and the all default icon sizes, throttled search, changed lodash import to include throttle only, merged container and presentational components, added card list logic selector, set correct key for card list entries, added Roman numeral to title of listed major cards, and added suit icon to all listed cards @fmapilis
